replica/table.cc

```diff
@@ -410,16 +410,16 @@ table::do_add_sstable(lw_shared_ptr<sstables::sstable_set> sstables, sstables::s
     auto new_sstables = make_lw_shared<sstables::sstable_set>(*sstables);
     new_sstables->insert(sstable);
     if (backlog_tracker) {
-        add_sstable_to_backlog_tracker(_compaction_strategy.get_backlog_tracker(), sstable);
+        table::add_sstable_to_backlog_tracker(get_backlog_tracker(), sstable);
```

Comment:
Ah, I see some difficulty with the idea. If compaction_manager becomes responsible, this code will have to be moved there.

Reply:
I think it's harder to move the responsibility to compaction_manager, because the tracker has to be renewed whenever compaction strategy changes. Previously, the tracker was stored in compaction_strategy_impl, which would be automatically renewed on strategy change. 

In this series, now tracker is moved to compaction_group. On strategy change, table calls each group with the new strategy so it can have a new tracker. Another good reason for it to live in compaction_group is that on split/merge, new trackers are automatically created for new groups, and automatically deleted for deleted groups.

Also, with updates to tracker done in compaction_group, we're less likely to see problems where tracker misses some updates to sstable set as it happened in the past, ex: https://github.com/scylladb/scylladb/issues/6441